app/src/renderer/features/session/sessionSlice.ts

```diff
@@ -0,0 +1,36 @@
+import { createSlice } from "@reduxjs/toolkit";
+import type { PayloadAction } from "@reduxjs/toolkit";
+import type { RootState } from "../../store";
+
+interface SessionState {
```

Comment:
maybe instead it makes for SessionState to be:
```
enum SessionStatus {
    Unauth,
    Offline,
    Auth
}

interface AuthData {
    expiration: string;
    token: string;
    journalistUUID: string;
    journalistFirstName: string;
    jouranlistLastName: string;
}

interface SessionState {
    status: SessionStatus;
    authData?: AuthData;   
}
```

so then we can explicitly check the `sessionState.status` (rather than if the journalist UUID is set), and can enforce that the `AuthData` is always fully populated if present.

Reply:
and then for the exposed interface, could export `setSessionState`/`setOfflineMode`/`clear`

Reply:
I like it and I'm implementing it. Quick question about convention. Do you prefer `journalistUuid` or `journalistUUID`?

Reply:
personally I think I slightly prefer `journalistUUID` (evidently lol) but am good with either convention!